supplementary_content/views.py

```diff
@@ -107,20 +123,6 @@ def _get_category(tree, id):
     return None
 
 
-def _matches_sections(content, title, part, sections):
-    for section in content['sections']:
-        if section['section'] in sections and section['title'] == title and section['part'] == part:
-            return True
-    return False
-
-
-def _filter_content(tree, title, part, sections):
-    for category in tree:
-        new_content = [item for item in category['supplementary_content'] if _matches_sections(item, title, part, sections)]
-        category['supplementary_content'] = new_content
-        _filter_content(category['sub_categories'], title, part, sections)
-
-
```

Comment:
The structure required these duplicate checks. The hope is that the cognitive complexity can be reduced by only checking for the filters in one way.